spec/std/db/dummy_driver.cr

```diff
@@ -0,0 +1,181 @@
+require "spec"
+
+class DummyDriver < DB::Driver
+  def build_connection(db : DB::Database) : DB::Connection
+    DummyConnection.new(db)
+  end
+
+  class DummyConnection < DB::Connection
+    def initialize(db)
+      super(db)
+      @@connections ||= [] of DummyConnection
```

Comment:
Why `||=` instead of just `=`? Then you don't have to use `not_nil!` everywhere.